src/main/kotlin/org/web3j/evm/EmbeddedEthereum.kt

```diff
@@ -321,15 +332,15 @@ class EmbeddedEthereum(configuration: Configuration, private val operationTracer
             .nonce(nonce)
             .sender(Address.fromHexString(web3jTransaction.from))
             .to(Address.fromHexString(web3jTransaction.to))
-            .payload(BytesValue.fromHexString(web3jTransaction.data ?: BytesValue.EMPTY.toString()))
+            .payload(Bytes.fromHexString(web3jTransaction.data ?: Bytes.EMPTY.toString()))
             .value(Wei.of(UInt256.fromHexString(web3jTransaction.value ?: "0x00")))
             .signature(FAKE_SIGNATURE)
             .build()
 
         val worldState =
             protocolContext.worldStateArchive.getMutable(protocolContext.blockchain.chainHeadHeader.stateRoot).get()
         val worldStateUpdater = worldState.updater()
-        val result = protocolSchedule.getByBlockNumber(protocolContext.blockchain.chainHeadBlockNumber).transactionProcessor.processTransaction(
+        val result: TransactionProcessingResult = protocolSchedule.getByBlockNumber(protocolContext.blockchain.chainHeadBlockNumber).transactionProcessor.processTransaction(
```

Comment:
No need to define the `result` variable, I think. We can just return the result directly.